csrc/custom_all_reduce.cuh

```diff
@@ -362,7 +434,11 @@ class CustomAllreduce {
       // note: must share the base address of each allocation, or we get wrong
       // address
       if (cuPointerGetAttribute(&base_ptr,
+#if defined(USE_ROCM)
```

Comment:
Same idea here. I generally think it's cleaner to do this kind of ifdef dispatching when declaring the constant variable and not in the implementation of a function.